prov/shm/src/smr_signal.h

```diff
@@ -54,6 +55,7 @@ static void smr_handle_signal(int signum, siginfo_t *info, void *ucontext)
 				sock_name, entry) {
 		unlink(sock_name->name);
 	}
+	pthread_mutex_unlock(&ep_list_lock);
```

Comment:
You need to move this up and for sock_name_list use sock_list_lock

Reply:
I closed the ep_list_lock around the ep list and used the sock_list_lock around the sock_name_list. Is this correct?